### PR TITLE
Add playwright driver

### PR DESCRIPTION
**Pull request by ftes**
To get discussion kicked off.

### CODE REVIEW

File: lib/phoenix_test.ex [hunk 1]
```diff
@@ -199,6 +199,11 @@ defmodule PhoenixTest do
   alias PhoenixTest.Driver
 
   @endpoint Application.compile_env(:phoenix_test, :endpoint)
+
+  def with_playwright(%Plug.Conn{} = conn, browser \\ :chromium) do
```
Comment (ftes) [inline lib/phoenix_test.ex:203]: Which browser should we run with?
- customise: global config, per-test options
  - e.g. user configures default browser globally, and can override per test
- allow running single test with 1..3 browsers

File: lib/phoenix_test/playwright.ex [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  alias PhoenixTest.Html
+  alias PhoenixTest.OpenBrowser
+  alias PhoenixTest.Query
+
+  @endpoint Application.compile_env(:phoenix_test, :endpoint)
+
+  defstruct page: nil, within: :none, last_field_query: :none
+
+  def build(_conn, path, browser_name) do
+    {:ok, browser} = Playwright.launch(browser_name)
```
Comment (ftes) [inline lib/phoenix_test/playwright.ex:14]: Not using `PlaywrightTest.Case`.
Which means we are missing teardown of the browser atm.

File: test/phoenix_test/static_test.exs [hunk 3]
```diff
@@ -4,12 +4,14 @@ defmodule PhoenixTest.StaticTest do
...
-    %{conn: Phoenix.ConnTest.build_conn()}
+  setup context do
+    conn = Phoenix.ConnTest.build_conn()
+    conn = if context[:playwright], do: with_playwright(conn), else: conn
+    %{conn: conn}
   end
 
   describe "render_page_title/1" do
-    test "renders the page title", %{conn: conn} do
+    test_also_with_playwright "renders the page title", %{conn: conn} do
```
Comment (ftes) [inline test/phoenix_test/static_test.exs:14]: Working test 🥳

File: mix.exs [hunk 4]
```diff
@@ -46,9 +46,11 @@ defmodule PhoenixTest.MixProject do
       {:jason, "~> 1.4"},
       {:phoenix, "~> 1.7.10"},
       {:phoenix_live_view, "~> 0.20.1"},
-      {:esbuild, "~> 0.8", only: :test, runtime: false},
+      {:esbuild, "~> 0.8", runtime: false},
```
Comment (ftes) [inline mix.exs:49]: `playwright` forces `esbuild` not only for test.

**Comment by ftes** [comment]
Wdyt: Is the time ripe to begin work on this @germsvel?

**Comment by germsvel** [comment]
> Wdyt: Is the time ripe to begin work on this @germsvel?

Thanks for asking @ftes. I have a branch that does some of the same things you do here, though some are different. I didn't open a draft PR because I don't want to give the impression that this is somehow almost ready. But the work is happening! Just gotta hang in there for me. Spending as much time as I can on it. 

That being said, happy to leave this draft open if it helps you. But I'm still exploring how best to introduce this separately.

**Comment by ftes** [comment]
I'll just ahead and close this then 👍

**Pull request closed without merging**